src/plugins/dna_stat/src/DNAStatPlugin.cpp

```diff
@@ -92,6 +93,13 @@ void DNAStatMSAEditorContext::buildStaticOrContextMenu(GObjectView *v, QMenu *m)
 void DNAStatMSAEditorContext::sl_showMSAProfileDialog() {
     GObjectViewAction *viewAction = qobject_cast<GObjectViewAction *>(sender());
     MSAEditor *msaEd = qobject_cast<MSAEditor *>(viewAction->getObjectView());
+    if (msaEd->getAlignmentLen() >= GRID_PROFILE_LENGTH_LIMIT) {
```

Comment:
Here is the best solution I see here:
1) If the profile is too long: force "Save to file" option to be ON.
2) Show warning that "Profile is too large and won't be opened in UGENE" and do not even try opening it.

This is much better than saying that the next user step will potentially "crash" UGENE: there are no crash at all here.

More details:
- The warning should be shown not as a separate annoying popup, but as a context label at the bottom of the dialog.
- You are right that opening/browsing very large profiles in UGENE viewer has no value. So as an optional extension you can even add a check box "Open profile in UGENE" which can be disabled for large alignments.